Sources/Datadog/Traces/Utils/Warnings.swift

```diff
@@ -0,0 +1,26 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2019-2020 Datadog, Inc.
+ */
+
+import Foundation
+
+/// Returns `true` if the warning was raised. `false` otherwise.
+internal func warn(if condition: @autoclosure () -> Bool, message: String) -> Bool {
+    if condition() { // TODO: RUMM-340 Consider thread safety
+        userLogger.warn(message)
+        return true
+    } else {
+        return false
+    }
+}
+
+/// Returns `nil` if the warning was raised. `T` otherwise.
+internal func warnIfCannotCast<T>(value: Any) -> T? {
+    guard let castedValue = value as? T else {
+        userLogger.warn("🔥 Using \(type(of: value.self)) while \(T.self) was expected.")
```

Comment:
as per [`printGenericInfo` example from Apple docs](https://developer.apple.com/documentation/swift/2885064-type), we should do the following instead:
```swift
userLogger.warn("🔥 Using \(type(of: value as Any)) while \(T.self) was expected.")
```

Reply:
Nice 💪 , didn't know that. Done ✅.